.ci/install_kata_kernel.sh

```diff
@@ -66,20 +86,89 @@ install_prebuilt_kernel() {
 	info "Installed pre-built kernel"
 }
 
-main() {
-	clone_katacontainers_repo
-	kernel_version=$(get_version "assets.kernel.version")
+install_kernel() {
+	local kernel_type=${1:-}
+	local kernel_version=${2:-}
+
+	info "Installing '${kernel_type}' kernel"
+
 	kernel_version=${kernel_version#v}
-	kata_config_version=$(cat "${kernel_packaging_dir}/kata_config_version")
-	kata_kernel_version="${kernel_version}-${kata_config_version}"
-	cached_kernel_version=$(curl -sfL "${latest_build_url}/latest") || cached_kernel_version="none"
-	info "Kata guest kernel : ${kata_kernel_version}"
+	local latest_build_url="${jenkins_url}/job/kata-containers-2.0-kernel-${kernel_type}-$(arch)-nightly/${cached_artifacts_path}"
+	local kata_config_version=$(cat "${kernel_packaging_dir}/kata_config_version")
+	local kata_kernel_version="${kernel_version}-${kata_config_version}"
+	local cached_kernel_version=$(curl -sfL "${latest_build_url}/latest") || cached_kernel_version="none"
+
+	info "Kata guest experimental kernel : ${kata_kernel_version}"
 	info "cached kernel  : ${cached_kernel_version}"
 
-	if ! install_prebuilt_kernel; then
+	# Due to bug https://github.com/kata-containers/kata-containers/issues/3145, it's not possible to
+	# build a kernel on fedora 35
+	if [[ "${ID}" == "fedora" ]] && [[ ${VERSION_ID} == 35 ]]; then
+		install_prebuilt_kernel ${latest_build_url} ${cached_kernel_version} ||
+			die "cannot build a kernel in this system. bug: https://github.com/kata-containers/kata-containers/issues/3145"
+	fi
+
+	if [[ "${kata_kernel_version}" != "${cached_kernel_version}" ]] ||
+		   ! install_prebuilt_kernel ${latest_build_url} ${cached_kernel_version}; then
 	    info "failed to install cached kernel, trying to build from source"
-	    build_and_install_kernel
+	    build_and_install_kernel "${kernel_type}" "${kernel_version}"
 	fi
 }
 
-main
+usage() {
+	exit_code="$1"
+	cat <<EOT
+Overview:
+
+	Build and install a kernel for Kata Containers
+
+Usage:
+
+	$script_name [options]
+
+Options:
+    -d          : Enable bash debug.
+    -h          : Display this help.
+    -t <kernel> : kernel type, such as vanilla, experimental, tdx.
```

Comment:
Idea: It's a bit more work, but you could consider adding a `-l` option to list the available types by using a hash of `type=>key:description`. Something like:

```bash
 kernel_types=(
    [experimental]="assets.kernel-experimental.tag:Experimental kernel"
    [tdx]="assets.kernel.tdx.tag:an Intel TDX (Trust Domain Extensions) kernel"

    # ...
  )
```